### PR TITLE
Use new registerApp endpoint

### PR DESCRIPTION
**Pull request by hpx7**

### CODE REVIEW

**Comment by dsiddharth** [comment]
@hpx7 when I run the code, it doesn't use the `APP_ID` env variable when connecting to the coordinator:
```
Connected to coordinator at coordinator.hathora.dev with appId 13e8cf44ea582c038798a7083e0cc8719abbe6a476d8c2855028bac60a44f8e8 and storeId 05847ead-de77-4086-826c-7e83ea94172e
```

Can you fix this to use `APP_ID` when initiating the connection? It's just a JSON object that can be easily augmented and the coordinator already knows how to handle it.

**Comment by hpx7** [comment]
Should be fixed already https://github.com/hathora/server-sdk-ts/commit/aaf3b337b106d1e250b2231db8b0ea3c4e45affb

**Comment by dsiddharth** [comment]
Yep, you're right. Just had to clear my gitignored file and it works as expected now:
```
Connected to coordinator at coordinator.hathora.dev with appId app-b125da32-5db8-4659-b0cf-28679f643146 and storeId 7851ad16-1b6c-4190-8b6c-44df4ce0f32d
Starting frontend at /Users/sdhulipalla/code/hathora/examples/pong/client/prototype-ui

> start
> vite


  vite v2.9.9 dev server running at:

  > Local:    http://localhost:3000/
  > Network:  http://172.20.10.11:3000/

  ready in 209ms.
```

File: src/commands/build.ts [hunk 1]
```diff
@@ -23,10 +23,10 @@ const cmd: CommandModule = {
           `${chalk.red("first")}`
       );
     } else {
```
Comment (dsiddharth) [inline src/commands/build.ts]: do this as a `return`

File: src/utils.ts [hunk 2]
```diff
@@ -64,16 +62,26 @@ export function getDirs() {
   };
 }
 
-export function getAppConfig() {
-  const appSecret = process.env.APP_SECRET?.trim() ?? uuidv4();
-  const appId = createHash("sha256").update(appSecret).digest("hex");
-  return { appId, appSecret };
+export async function getAppConfig() {
+  const res = await axios.post<{ appId: string; appSecret: string }>("https://coordinator.hathora.dev/registerApp");
```
Comment (dsiddharth) [inline src/utils.ts]: make this configurable

File: src/utils.ts [hunk 3]
```diff
@@ -64,16 +62,26 @@ export function getDirs() {
...
-export function generateLocal() {
+export async function generateLocal() {
   const { rootDir } = getDirs();
-  dotenv.config({ path: join(rootDir, ".env") });
-  const appConfig = getAppConfig();
+
+  let appConfig: { appId: string; appSecret: string };
+  if (!existsSync(join(rootDir, ".env"))) {
+    appConfig = await getAppConfig();
+    outputFileSync(join(rootDir, ".env"), `APP_ID=${appConfig.appId}\nAPP_SECRET=${appConfig.appSecret}\n`);
```
Comment (dsiddharth) [inline src/utils.ts:82]: parse this, check if file doesn't exist or APP_ID isn't present, generate a new one

**Review by dsiddharth — CHANGES_REQUESTED** [review]

**Review by dsiddharth — APPROVED** [review]

**Pull request merged**